### DISCUSSION TITLE
DateTimePicker timezone always jumps to Jan 31 for startOfMonth

### DISCUSSION THREAD

**Discussion by madsem** [filamentphp/filament#5600] [Help]
Trying to build a DateTimePicker with timezones, because in our app we have data in several different timezones that need to be normalized when being displayed.

```php
public function getTimezoneConfiguredDatePickerCombo(
        string $timezone = 'America/New_York',
        string $range = 'month'
    ): array {

        DateTimePicker::configureUsing(fn(DateTimePicker $component) => $component->timezone($timezone));
        $now = Carbon::now()->setTimezone($timezone);

        $startDate = ($range == 'month') ? $now->copy()->startOfMonth() : $now->copy();
        $endDate = ($range == 'month') ? $now->copy()->endOfMonth() : $now->copy();

        return [
            Grid::make()
                ->schema([
                    DateTimePicker::make('date_from')
                        ->default($startDate)
                        ->format('yyyy-dd-mm')
                        ->withoutTime()
                        ->weekStartsOnMonday()
                        ->closeOnDateSelection(),
                    DateTimePicker::make('date_until')
                        ->default($endDate)
                        ->format('yyyy-dd-mm')
                        ->withoutTime()
                        ->weekStartsOnMonday()
                        ->closeOnDateSelection(),
                ]),
        ];
    }
```

Expected result:
DateTimePicker # 1 displays "Febr. 1"
DateTimePicker # 2 displays "Febr. 28"

Given Result:
DateTimePicker # 1 displays "Jan 31"
DateTimePicker # 2 displays "Febr. 28"

When I click "reset filters" the correct dates are displayed, but as soon as the page is reloaded it jumps back to Jan 31 - Feb 28.
I am saving filter state with `shouldPersistTableFiltersInSession`, I set this to false but it didn't change anything.

What am I doing wrong here? Or is this a bug

**Comment by danharrin** [comment]
Before passing the date to the field, convert it to UTC?

**Reply by madsem** [reply]
Doesn't seem to help, changed `$now = Carbon::now()->setTimezone('UTC');` for testing.
But now it just stays at "Jan 31" in the first DateTimePicker, even when resetting filters.

**Reply by danharrin** [reply]
worth opening an issue with a very very simple reproduction repository

**Reply by madsem** [reply]
Will try to whip that up. When changing the timezone in the method argument to utc `string $timezone = 'UTC'`,
it works as expected.

So when both `Carbon` instance and `DateTimePicker` are configured using our app default timezone (UTC), it works without problems. But as soon as I change the timezone, the issue comes back.

**Reply by madsem** [reply]
Created a repo https://github.com/madsem/filamentphp-datetimepicker
Will open the issue now, thanks! :)

**Comment by photojo** [comment]
Is it possible with `DateTimePicker` to set time only, f. e. to 07.00?
